src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    private void deleteDriverFromCar(Long carId, Long driverId) {
+        String query = "DELETE FROM cars_drivers WHERE driver_id = ? AND car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            statement.setLong(2, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete drivers for car by id "
+                    + carId + " in carsDB.", e);
+        }
+    }
+
+    private void insertDriverToCar(Long carId, Long driverId) {
+        String query = "INSERT INTO cars_drivers (driver_id, car_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            statement.setLong(2, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert drivers for car by id "
+                    + carId + " in carsDB.", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id "
+                + "JOIN cars c ON c.id = cd.car_id "
+                + "WHERE c.is_deleted = FALSE AND d.id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars "
+                    + "for driver by id " + driverId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        ManufacturerServiceImpl manufacturerService =
```

Comment:
You don't have to call service methods at the Dao level, do the joins directly in the request.